### DISCUSSION TITLE
Help for MQ2 Sensors

### DISCUSSION THREAD

**Discussion by NahianMugdho** [micropython/micropython#9995] [Libraries & Drivers]
Hello, I'm using the following library for MQ-2 sensors - [https://github.com/kartun83/micropython-MQ](https://github.com/kartun83/micropython-MQ) 
I'm using ESP-8266 in this case. Here is my code- 
```
from machine import Pin,ADC,Timer

from MQ2 import MQ2
                       

met=MQ2(0)
val = met.readLPG()
print(val)
```

I'm getting the following errors-

> Traceback (most recent call last):
>   File "<stdin>", line 8, in <module>
>   File "MQ2.py", line 18, in readLPG
>   File "BaseMQ.py", line 150, in readScaled
> ValueError: math domain error

I couldn't find out any specific example about this library. Please help me out.

**Comment by jimmo** [comment]
@NahianMugdho You can follow this through the code: `math domain error` in `readScaled` means that it tried to do `math.log(0)`. If you trace `readLPG` to `readScaled`, this means that `readRatio()` must be returning zero. This means that `__readRs__` is returning zero, which means `self.__calculateResistance__(self.pinData.read())` is returning zero.

If you read the `__calculateResistance__` function, the only way that returns zero is when the ADC value is 1023.

Suspect there's something wrong with your wiring and the ADC isn't reading the right pin from the sensor.

Note: you might also need to configure baseVoltage to 3.3 if you're powering the MQ2 from the ESP8266. e.g.

```
met=MQ2(0, baseVoltage=3.3)
```
